app/controllers/concerns/two_factor_authenticatable_methods.rb

```diff
@@ -157,6 +157,22 @@ def invalid_otp_error(type)
     end
   end
 
+  def attempts_remaining_warning
+    if attempts_count_remaining < IdentityConfig.store.minimum_attempts_remaining_warning_count
+      t(
+        'two_factor_authentication.attempt_remaining_warning_html',
+        count: attempts_count_remaining,
+      )
+    else
+      ''
+    end
+  end
+
+  def attempts_count_remaining
+    IdentityConfig.store.login_otp_confirmation_max_attempts -
```

Comment:
Since this concern is generalized to all MFA, it'd be nice if we could move the OTP-specific handling elsewhere. Then again, since `invalid_otp_error` is already handling the MFA-specific messages and this is essentially an extension of that, maybe it's not an issue.

Could be worth clarifying the name of the method that this is specific to OTP though.